### PR TITLE
Install and start logging agent based on kube env

### LINKED ISSUE #81299: InstallAndStart-LoggingAgent should be based on whether logging is enabled for the cluster
**What happened**:

The stackdriver logging agent is installed and start running no matter logging is enabled/disabled in the cluster or not.

**What you expected to happen**:

Do not install and start the logging agent if logging is disabled.

### PR DESCRIPTION
**Pull request by liyanhui1228**
**What type of PR is this?**

/kind bug

**Which issue(s) this PR fixes**:

Fixes #81299

```release-note
Install and start logging agent based on kube env ENABLE_NODE_LOGGING.
```

### CODE REVIEW

File: cluster/gce/windows/configure.ps1 [hunk 1]
```diff
@@ -109,10 +123,13 @@ try {
     StartProcess-WriteSshKeys
   }
 
+  if (IsLoggingEnabled $kube_env) {
+    InstallAndStart-LoggingAgent
```
Comment (yujuhong) [inline cluster/gce/windows/configure.ps1]: @pjh if stackdriver is already installed on the image, would it be started automatically?

File: cluster/gce/windows/configure.ps1 [hunk 2]
```diff
@@ -84,6 +84,20 @@ function FetchAndImport-ModuleFromMetadata {
...
 }
 
+# Returns true if the ENABLE_NODE_LOGGING field in kube_env is true.
+# $KubeEnv is a hash table containing the kube-env metadata keys+values.
+function IsLoggingEnabled {
+  param (
+    [parameter(Mandatory=$true)] [hashtable]$KubeEnv
+  )
+
+  if ($KubeEnv.Contains('ENABLE_NODE_LOGGING') -and `
```
Comment (yujuhong) [inline cluster/gce/windows/configure.ps1]: Is this enabled by default?

Comment (liyanhui1228) [reply cluster/gce/windows/configure.ps1]: Yes it is.

File: cluster/gce/windows/configure.ps1 [hunk 3]
```diff
@@ -84,6 +84,20 @@ function FetchAndImport-ModuleFromMetadata {
...
+      ($KubeEnv['ENABLE_NODE_LOGGING'] -eq 'true')) {
+    return $true
```
Comment (pjh) [inline cluster/gce/windows/configure.ps1]: I see that this currently looks like https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/windows/common.psm1#L532, but can this just be?:

```
  return ($KubeEnv.Contains('ENABLE_NODE_LOGGING') -and `
      ($KubeEnv['ENABLE_NODE_LOGGING'] -eq 'true'))
```

Mostly just curious, if you don't want to bother trying it that's fine.

File: cluster/gce/windows/configure.ps1 (hunk 1 above)
Comment (pjh) [reply cluster/gce/windows/configure.ps1]: Yes. https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/windows/k8s-node-setup.psm1#L1178

@liyanhui1228 could you update the comment above that function to state that even if Stackdriver is already installed, the function will still [re]start the service?

File: cluster/gce/windows/configure.ps1 (hunk 3 above)
Comment (pjh) [inline cluster/gce/windows/configure.ps1]: Could this just be?:

```
  return ($KubeEnv.Contains('ENABLE_NODE_LOGGING') -and `
     ($KubeEnv['ENABLE_NODE_LOGGING'] -eq 'true'))
```

I see this is imitating code that we already have; I'm mostly just curious, if you don't want to bother trying this that's fine.

**Review by pjh — APPROVED** [review]

File: cluster/gce/windows/configure.ps1 (hunk 1 above)
Comment (liyanhui1228) [reply cluster/gce/windows/configure.ps1]: Done.

File: cluster/gce/windows/configure.ps1 (hunk 3 above)
Comment (liyanhui1228) [reply cluster/gce/windows/configure.ps1]: Yeah this is imitating the existing code and thanks for the suggestion! I've added a new kube_env `ENABLE_STACKDRIVER_WINDOWS` as the `ENABLE_NODE_LOGGING` will be deprecated soon. And I just kept them in the if-elseif.

**Comment by yujuhong** [comment]
The change itself looks good to me, but seems like you need to clean up your commits :-)

**Comment by liyanhui1228** [comment]
Done, I just rebased the branch.

**Comment by yujuhong** [comment]
This is a bug fix for GCE/Windows only, and should not affect other environments.

**Pull request merged**